src/config.rs

```diff
@@ -118,6 +118,45 @@ impl AssignConfig {
     }
 }
 
+#[derive(PartialEq, Eq, Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub(crate) enum WarnNonDefaultBranchConfig {
+    Simple(bool),
+    Extended {
+        enable: bool,
+        /// List of exceptions that have a different default branch
+        #[serde(default)]
+        exceptions: Vec<WarnNonDefaultBranchException>,
```

Comment:
We could also use a `HashMap<String, WarnNonDefaultBranchException>` where the string would be the branch name, which would permit this and be more consistent with the rest of the configs, but we would losz the "exceptions" field. I'm not sure which one is better.
```toml
[assign.warn_non_default_branch]
enable = true

[assign.warn_non_default_branch."beta"]
title = "[beta"

[assign.warn_non_default_branch."stable"]
title = "[stable"
```